packages/core/src/entity/asset/asset.entity.ts

```diff
@@ -44,4 +45,8 @@ export class Asset extends VendureEntity implements Taggable {
     @ManyToMany(type => Tag)
     @JoinTable()
     tags: Tag[];
+
+    @ManyToMany(type => Channel)
+    @JoinTable()
+    channels: Channel[];
```

Comment:
> If a Product has assets not in the current channel, they are still returned in the `Product` type. They should be filtered out in the field resolver, in the same way we do with [facetValues](https://github.com/vendure-ecommerce/vendure/blob/e8fcb9959ab0717af087398b075ee357e467f1b4/packages/core/src/api/resolvers/entity/product-entity.resolver.ts#L67-L76). An e2e test for this would be good too.

Ahh yes, agreed. Should this be solved in the `asset.resolver` or in the `asset.service` ?

Reply:
I just noticed that the resolver _always_ gets the data from the AssetService, so in this case it's probably better to do it directly in the service, as a DB-based solution would be slightly more efficient.

Reply:
I am struggling a bit with making this testable:
If an Asset belongs to a Product, and we remove the Product from a channel, the assets are also removed from that channel. How could we end up with a Product, that has assets which don't belong to the channel?

Reply:
1. Add a product from default into channel 2.
2. Switch to default channel.
3. Add an Asset "A" (which is only available in default channel) to the product.
4. Switch to channel 2 and query the product with assets.
5. Assert that asset "A" is not in the product.assets array.

Reply:
I am having trouble getting the `asset.asset.channels` relation on ProductAsset. I created a seperate issue for this case: #717